javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -104,7 +99,7 @@ public static boolean areSame(
             if (a1 == a2) {
                 return true;
             }
-            if (annotationName(a1) != annotationName(a2)) {
+            if (!annotationName(a1).equals(annotationName(a2))) {
```

Comment:
@wmdietl Could we compare by `a1.getAnnotationType() == a2.getAnnotatedType()` instead of by name?